docker-compose.yml

```diff
@@ -1,74 +1,86 @@
-# Reference for exteranl_links and bridge containers
-# https://blog.virtualzone.de/2016/09/docker-compose-link-containers-outside-compose-file-using-external_links.html
-
-version: '2'
+version: '3'
 services:
-  base:
-    build: .
-    image: promgen
-    command: sh -c "promgen migrate && promgen register docker-demo localhost 9090"
-    environment:
-      CELERY_BROKER_URL: redis://redis:6379/0
-      DATABASE_URL: mysql://promgen:promgen@mysql:3306/promgen
-      SECRET_KEY: asdf
+  # Promgen services
 
   web:
-    extends: base
-    command: promgen runserver 0.0.0.0:8000
-    ports:
-      - "8080:8000"
-    links:
+    image: line/promgen
+    depends_on:
       - mysql
       - redis
     environment:
       DEBUG: 1
+      CELERY_BROKER_URL: redis://redis:6379/0
+      DATABASE_URL: mysql://root:secretrootpassword@mysql:3306/promgen
+      SECRET_KEY: some-secret-key-AhvSh+xGkm9kA,ghfi4?w8K
+    command: wait-runserver 0.0.0.0:8000
+    volumes:
+      - ./promgen:/etc/promgen
+    ports:
+      - "8000:8000"
 
   worker:
-    extends: base
-    command: worker -l info --queues localhost,celery
-    links:
-      - prometheus
+    image: line/promgen
+    depends_on:
       - mysql
       - redis
+    environment:
+      CELERY_BROKER_URL: redis://redis:6379/0
+      DATABASE_URL: mysql://root:secretrootpassword@mysql:3306/promgen
+      SECRET_KEY: some-secret-key-AhvSh+xGkm9kA,ghfi4?w8K
+    command: worker -l info --queues celery,prometheus
     volumes:
-      - ./docker:/etc/prometheus
-      - ./docker/prom_data:/prometheus
+      - ./prometheus:/etc/prometheus
+      - ./promgen:/etc/promgen
+
+  # Promgen setup
+
+  promgen-setup:
+    image: line/promgen
+    depends_on:
+      - mysql
+    environment:
+      PROMGEN_REGISTER_SHARD: docker-default
+      PROMGEN_REGISTER_HOST: prometheus
+      PROMGEN_REGISTER_PORT: 9090
+      DATABASE_URL: mysql://root:secretrootpassword@mysql:3306/promgen
+      SECRET_KEY: some-secret-key-AhvSh+xGkm9kA,ghfi4?w8K
```

Comment:
I'm not familiar with django commands and best practices, should this stanza also include `DEBUG: 1`, the `CELERY_BROKER_URL`? Is `SECRET_KEY` even required?

Reply:
DEBUG would only be necessary for times when you want extra debugging information, typically just for development. Could drop it from the docker-compose file and be fine.

SECRET_KEY in theory is only required for http but I believe the django startup code errors out if it is not set somewhere. Could set a default SECRET_KEY again in settings.py so that we could remove it from the docker-compose file.